tests/Unit/Evolution/DgSubcell/Actions/Test_TciAndRollback.cpp

```diff
@@ -0,0 +1,427 @@
+                comp,
+                SelfStart::Tags::InitialValue<Tags::Variables<prim_vars_tags>>>(
+                runner, 0));
+        CHECK(initial_value_prim_vars_from_box ==
+              evolution::dg::subcell::fd::project(
+                  initial_value_prim_vars, dg_mesh, subcell_mesh.extents()));
+      }
+    }
+  } else {
+    CHECK(ActionTesting::get_next_action_index<comp>(runner, 0) == 2);
+    CHECK(active_grid_from_box == evolution::dg::subcell::ActiveGrid::Dg);
+    CHECK(nhbr_data_from_box.empty());
+    CHECK_FALSE(did_rollback_from_box);
+
+    CHECK(time_stepper_history_from_box.size() == history_size);
+    CHECK(time_stepper_history_from_box.integration_order() ==
+          time_stepper_history.integration_order());
+    for (auto expected_it = time_stepper_history.begin(),
+              it = time_stepper_history_from_box.begin();
+         expected_it != time_stepper_history.end(); ++it, ++expected_it) {
+      CHECK(it.time_step_id() == expected_it.time_step_id());
+      CHECK(it.value() == expected_it.value());
+      CHECK(it.derivative() == expected_it.derivative());
+    }
+
+    CHECK(get(get<evolution::dg::subcell::Tags::Inactive<Var1>>(
+              inactive_vars_from_box)) ==
+          evolution::dg::subcell::fd::project(get(get<Var1>(evolved_vars)),
+                                              dg_mesh, subcell_mesh.extents()));
+    CHECK(active_vars_from_box == evolved_vars);
+    if (self_starting) {
+      CHECK(initial_value_evolved_vars_from_box == initial_value_evolved_vars);
+      if constexpr (HasPrims) {
+        const auto& initial_value_prim_vars_from_box = get<0>(
+            ActionTesting::get_databox_tag<
+                comp,
+                SelfStart::Tags::InitialValue<Tags::Variables<prim_vars_tags>>>(
+                runner, 0));
+        CHECK(initial_value_prim_vars_from_box == initial_value_prim_vars);
+      }
+    }
+  }
+}
+
+template <size_t Dim>
+void test() {
+  for (const bool rdmp_fails : {true, false}) {
+    for (const bool tci_fails : {false, true}) {
+      for (const bool always_use_subcell : {false, true}) {
+        for (const bool self_starting : {false, true}) {
+          test_impl<Dim, true>(rdmp_fails, tci_fails, always_use_subcell,
+                               self_starting);
+          test_impl<Dim, false>(rdmp_fails, tci_fails, always_use_subcell,
+                                self_starting);
+        }
+      }
+    }
+  }
+}
+
+SPECTRE_TEST_CASE("Unit.Evolution.Subcell.Actions.TciAndRollback",
+                  "[Evolution][Unit]") {
+  // 1. Test RDMP passes/fails (check TciMutator not called on failure)
+  // 2. Test always_use_subcells (maybe?)
+  // 3. Test TciMutator passes/fails
+  //
+  // - history projected with correct size (one fewer)
+  // - active vars become projection of latest in history
+  // - active_grid is correct
+  // - did_rollback is correct
+  // - if self-start check initial value (and prims) were projected
```

Comment:
I don't understand the relationship of the first numbered list (lines 414-16) and this list (418-422). Is one a sublist of the other? If not, why are there two lists? Some pre-list sentences describing what is being listed out would be very helpful here...